workers/unity/Packages/com.improbable.gdk.core/Subscriptions/CalllbackManagers/ResponseCallbackManager.cs

```diff
@@ -20,10 +20,10 @@ public ResponseCallbackManager(World world)
         public void InvokeCallbacks()
         {
             var responses = commandSystem.GetResponses<T>();
-            foreach (var response in responses)
+            for (int i = 0; i < responses.Count; ++i)
             {
-                callbacks.InvokeAll(response.GetRequestId(), response);
-                callbacks.RemoveAllCallbacksForIndex(response.GetRequestId());
+                callbacks.InvokeAll(responses[i].GetRequestId(), responses[i]);
```

Comment:
Put `responses[i]` into a variable before using to avoid indexing multiple times.

Reply:
Sure. Although I'd be very surprised if this isn't optimised out.

Reply:
The bigger performance problem here is the callbacks doing things by value probably. But that change can be made later.